File: 2428-equal-row-and-column-pairs/2428-equal-row-and-column-pairs.cpp
class Solution {
struct VectorHash {
    size_t operator()(const vector<int>& vec) const {
        size_t hash = 0;
        for (int num : vec) {
            hash ^= hash * 31 + hash + num;
        }
        return hash;
    }
};
public:
    int equalPairs(vector<vector<int>>& grid) {
        unordered_map<vector<int>, int, VectorHash> mp; // Use custom hash function
        int n = grid.size();

        // Count the frequency of each row
        for (const auto& row : grid) {
            mp[row]++;
        }

        // for each columns..
        int res = 0;
        for (int col = 0; col < n; ++col) {
            vector<int> column;
            for (int row = 0; row < n; ++row) {
                column.push_back(grid[row][col]);
            }
            if (mp.count(column)) {
                res += mp[column];
            }
        }

        return res;
    }
};

File: 2428-equal-row-and-column-pairs/2428-equal-row-and-column-pairs.py
class Solution:
    def equalPairs(self, grid: List[List[int]]) -> int:
        
        mp = {}
        t = tuple()

        for r in grid:
            t = tuple(r)
            mp[t] = mp.get(t, 0) + 1

        grid = zip(*grid) # transpose grid
        
        res = 0
        for row in grid:
            if row in mp:
                res += mp[row]

        return res